src/middlewared/middlewared/plugins/account.py

```diff
@@ -1638,6 +1638,33 @@ async def common_validation(
                         f'{entry["bsdgrp_group"]}: membership of this builtin group may not be altered.'
                     )
 
+        # Root user restrictions
+        if old and old['uid'] == 0:  # Root user being updated
+            if 'groups' in data:
+                # Get builtin_administrators group ID
+                builtin_admin_group = await self.middleware.call(
+                    'group.query',
+                    [('group', '=', 'builtin_administrators'), ('local', '=', True)],
+                    {'get': True}
+                )
+                builtin_admin_gid = builtin_admin_group['id']
```

Comment:
That's not a gid. It's the datastore primary key. Correct field, but incorrectly labelled.